File: 692-top-k-frequent-words/top-k-frequent-words.java
class Solution {
    public List<String> topKFrequent(String[] words, int k) {
        List<String> ans = new ArrayList<>();
        Map<String,Integer> mp = new TreeMap<>();
        int n = words.length;
        for(String s : words)
        {
            mp.put(s,mp.getOrDefault(s,0)+1);
        }
        List<String>[] temp= new ArrayList[n];
        for(Map.Entry<String, Integer> entry : mp.entrySet())
        {
            int kk = entry.getValue();
            String s = entry.getKey();
            if(temp[kk]==null) temp[kk]= new ArrayList<>();
            temp[kk].add(s);
        }
        for(int i = temp.length-1;i>-1;i--)
        {
            if(temp[i]!=null)
            {
                for( String s : temp[i])
                {
                    ans.add(s);
                    k--;
                    if(k==0) return ans;
                }
            }
        }
        return ans;
    }
}

File: 692-top-k-frequent-words/top-k-frequent-words.cpp
class Solution {
public:
    vector<string> topKFrequent(vector<string>& words, int k) {
        map<string,int> mp;
        vector<string> ans;
        for( string s : words)
        {
            mp[s]++;
        }
        vector<vector<string>> temp(words.size()+1);
        for(auto i = mp.begin() ;i!=mp.end();i++)
        {
            int freq = i->second;
            string s = i->first;
            temp[freq].push_back(s);
        }
        for(int i = words.size() ;i>-1;i--)
        {
            for(string s : temp[i])
            {
                  k--;
                  ans.push_back(s);
                  if(k==0) return ans;
            }
        }
        return ans;
    }
};